tools/topology/topology2/platform/intel/dmic-aec.conf

```diff
@@ -0,0 +1,363 @@
+               # num_pdm_active should always set to 2 but depending on the number of DMIC's
+               # the mic's are enabled or disabled in each PDM.
+               num_pdm_active  2
+
+               Object.Base.hw_config."DMIC1" {
+                       id      0
+               }
+
+               # PDM controller config
+               Object.Base.pdm_config."3" {
+                       mic_a_enable    $PDM0_MIC_A_ENABLE
+                       mic_b_enable    $PDM0_MIC_B_ENABLE
+                       ctrl_id 0
+               }
+               Object.Base.pdm_config."4" {
+                       ctrl_id 1
+                       mic_a_enable    $PDM1_MIC_A_ENABLE
+                       mic_b_enable    $PDM1_MIC_B_ENABLE
+               }
+       }
+}
+
+Object.Pipeline {
+	gain-capture.3 {
+		format		$FORMAT
+		index 		$DMIC0_HOST_PIPELINE_ID
+		Object.Widget.pipeline.1 {
+			stream_name $DMIC0_PIPELINE_STREAM_NAME
+		}
+		Object.Widget.copier.1 {
+			stream_name $DMIC0_PCM_CAPS
+			num_audio_formats 2
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+			Object.Base.audio_format.2 {
+				in_channels		4
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_channels		4
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+				in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				in_ch_map	$CHANNEL_MAP_3_POINT_1
+				out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				out_ch_map	$CHANNEL_MAP_3_POINT_1
+			}
+		}
+		Object.Widget.gain.1 {
+			num_audio_formats 2
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+			Object.Base.audio_format.2 {
+				in_channels		4
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_channels		4
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+				in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				in_ch_map	$CHANNEL_MAP_3_POINT_1
+				out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				out_ch_map	$CHANNEL_MAP_3_POINT_1
+			}
+			Object.Control.mixer.1 {
+				name 'DMIC0 Capture Volume 1'
+			}
+		}
+		Object.Widget.copier."2" {
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+			Object.Base.audio_format.2 {
+				in_channels		4
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_channels		4
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+				in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				in_ch_map	$CHANNEL_MAP_3_POINT_1
+				out_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+				out_ch_map	$CHANNEL_MAP_3_POINT_1
+			}
+		}
+	}
+
+	gain-capture.5 {
+		format		$FORMAT
+		index		$AEC_HOST_PIPELINE_ID
+		Object.Widget.pipeline.1 {
+			stream_name "AEC Feedback"
```

Comment:
this naming makes limited sense, I have no idea what this refers to.
for a nocodec topology we only don't need to refer to specific functionality, this is just a second capture.

Reply:
sure, thanks